pkg/kubectl/cmd/set/set_resources.go

```diff
@@ -109,7 +109,7 @@ func NewCmdResources(f cmdutil.Factory, streams genericclioptions.IOStreams) *co
 	o := NewResourcesOptions(streams)
 
 	resourceTypesWithPodTemplate := []string{}
-	for _, resource := range f.SuggestedPodTemplateResources() {
+	for _, resource := range o.SuggestedPodTemplateResources() {
```

Comment:
The list of resources that contain pod templates is not statically determinant.  Similar to `get` and `describe` and other commands that generic resources, we should suggest the use of `kubectl api-resources` instead.  See `SuggestApiResources` in pkg/kubectl/cmd/util .

Reply:
If so, then I think `suggestedPodTemplateResources` can be deleted. @juanvallejo WDYT?

Reply:
@wgliang Yep, I agree it should be deleted